app/controllers/empire/material.js

```diff
@@ -1,22 +1,25 @@
 import Controller from '@ember/controller';
 import { inject as controller } from '@ember/controller';
 import { computed } from '@ember/object';
-import { filter, lt, or } from '@ember/object/computed';
+import { mapBy, and, sum, filter, lt, or } from '@ember/object/computed';
 
 export default Controller.extend({
   empireCtl: controller('empire'),
   isGenMaterialOnCooldown: lt('model.spellPoints', 1),
   isGenMaterialDisabled: or('isGenMaterialOnCooldown', 'model.dead', 'empireCtl.isMaxMaterial'),
-  isGenMaterialAvailable: computed('model.type', function() {
-    return this.model.type == "religious"
-  }),
-  workerGathererAvailable: computed('model.type', 'game.upgrades.@each.isActive', function() {
-    return (this.model.type == "economical" || this.game.getUpgrade('Universal Worker').isActive)
-      // TODO: this.game.getUpgrade('Gathering').isActive &&
+  isGenMaterialAvailable: computed('model.type', 'game.upgrades.@each.isActive', function() {
+    return this.model.type == "religious" && this.game.getUpgrade('Magic Anvil').isActive
   }),
 
   materialStorageBuildings: filter('model.materialStorageBuildings', b => ! b.isCapital),
+  // This is ugly: Use sum to do a reduced `or`, because ember's functional sucks balls.
```

Comment:
Better: Do as in templates, and upgrade the filtering so that you can directly do `{{#if list}}`